presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -238,17 +239,21 @@ private RelationPlan addColumnMasks(Table table, RelationPlan plan)
                 for (Symbol symbol : plan.getRoot().getOutputSymbols()) {
                     assignments.put(symbol, symbol.toSymbolReference());
                 }
-                assignments.put(plan.getFieldMappings().get(i), coerceIfNecessary(analysis, mask, planBuilder.rewrite(mask)));
+
+                Symbol symbol = symbolAllocator.newSymbol(plan.getFieldMappings().get(i));
```

Comment:
> There's a latent bug in PredicatePushdownOptimizer when field names are reused
in derivative expressions. This fix allocates a unique symbol for each application
of a mask.

What it takes to fix `PredicatePushdownOptimizer` instead? Or maybe we should admit that each symbol name in the plan has to be unique, then there will be no bug in PPD (we already hit that case couple times). WDYT?

Reply:
Requiring unique symbol names would break many other assumptions. The purpose of UnaliasSymbolReferences is to remove unique names that refer tho the same thing.

What this change does is make the masks use unique names, as it appears to be a problem when it’s not a pure symbol rename.

I haven looked in depth at what it would take to fix PPD, but that class Has become too complex.